ledger-core/virtual/integration/method_test.go

```diff
@@ -348,867 +390,540 @@ func TestVirtual_CallMethodAfterPulseChange(t *testing.T) {
+			logger := inslogger.FromContext(ctx)
 
-		classB        = gen.UniqueGlobalRef()
-		objectBGlobal = reference.NewSelf(server.RandomLocalWithPulse())
+			executeDone := server.Journal.WaitStopOf(&execute.SMExecute{}, 2)
 
-		outgoingCallRef = gen.UniqueGlobalRef()
-	)
+			runnerMock := logicless.NewServiceMock(ctx, mc, func(execution execution.Context) string {
+				return execution.Request.CallSiteMethod
+			})
+			server.ReplaceRunner(runnerMock)
+			server.Init(ctx)
+			server.IncrementPulseAndWaitIdle(ctx)
 
-	Method_PrepareObject(ctx, server, payload.Ready, objectBGlobal)
+			var (
+				class = gen.UniqueGlobalRef()
 
-	// add ExecutionMocks to runnerMock
-	{
-		outgoingCall := execution.NewRPCBuilder(outgoingCallRef, objectAGlobal).CallMethod(objectBGlobal, classB, "Foo", []byte("123"))
-		objectAResult := requestresult.New([]byte("finish A.New"), objectAGlobal)
-		objectAResult.SetActivate(reference.Global{}, classA, []byte("state A"))
-		objectAExecutionMock := runnerMock.AddExecutionMock("New")
-		objectAExecutionMock.AddStart(
-			func(ctx execution.Context) {
-				logger.Debug("ExecutionStart [A.New]")
-				require.Equal(t, classA, ctx.Request.Callee)
-				require.Equal(t, outgoingA, ctx.Request.CallOutgoing)
-			},
-			&execution.Update{
-				Type:     execution.OutgoingCall,
-				Error:    nil,
-				Outgoing: outgoingCall,
-			},
-		)
-		objectAExecutionMock.AddContinue(
-			func(result []byte) {
-				logger.Debug("ExecutionContinue [A.New]")
-				require.Equal(t, []byte("finish B.Foo"), result)
-			},
-			&execution.Update{
-				Type:   execution.Done,
-				Result: objectAResult,
-			},
-		)
+				objectAGlobal = reference.NewSelf(server.RandomLocalWithPulse())
+				objectBGlobal = reference.NewSelf(server.RandomLocalWithPulse())
+			)
 
-		runnerMock.AddExecutionMock("Foo").AddStart(
-			func(ctx execution.Context) {
-				logger.Debug("ExecutionStart [B.Foo]")
-				require.Equal(t, objectBGlobal, ctx.Request.Callee)
-				require.Equal(t, objectAGlobal, ctx.Request.Caller)
-				require.Equal(t, []byte("123"), ctx.Request.Arguments)
-			},
-			&execution.Update{
-				Type:   execution.Done,
-				Result: requestresult.New([]byte("finish B.Foo"), objectBGlobal),
-			},
-		)
+			Method_PrepareObject(ctx, server, payload.Ready, objectAGlobal)
+			Method_PrepareObject(ctx, server, payload.Ready, objectBGlobal)
+
+			// add mock
+			outgoingCall := execution.NewRPCBuilder(gen.UniqueGlobalRef(), objectAGlobal).CallMethod(objectBGlobal, class, "Bar", []byte{})
+			objectAExecutionMock := runnerMock.AddExecutionMock("Foo")
+			objectAExecutionMock.AddStart(
+				func(ctx execution.Context) {
+					require.Equal(t, objectAGlobal, ctx.Request.Callee)
+					logger.Debug("ExecutionStart [A.Foo]")
+				},
+				&execution.Update{
+					Type:     execution.OutgoingCall,
+					Error:    nil,
+					Outgoing: outgoingCall,
+				},
+			)
+			objectAExecutionMock.AddContinue(
+				func(result []byte) {
+					logger.Debug("ExecutionContinue [A.Foo]")
```

Comment:
check result, please

Reply:
why need check in this place? and what is the fields need check?

Reply:
because test check calls from method to method, method args and result are important to check  
`result []byte` - this field

Reply:
done